LinqTask/Program.cs

```diff
@@ -0,0 +1,69 @@
+            Console.WriteLine(
+                "baaa;aabb;xabbx;abb;aaaaa;ccc;dap;zh".Split(';').Except
+                ("baaa;aabb;xabbx;abb;aaaaa;ccc;dap;zh".Split(';').Where(word=>word.EndsWith("bb")).Skip(2)).Last());
+
+
+
+            Console.WriteLine("======================LEVEL 2===========================");
+
+            var data = new List<object>() {
+                        "Hello",
+                        new Book() { Author = "Terry Pratchett", Name = "Guards! Guards!", Pages = 810 },
+                        new List<int>() {4, 6, 8, 2},
+                        new string[] {"Hello inside array"},
+                        new Film() { Author = "Martin Scorsese", Name= "The Departed", Actors = new List<Actor>() {
+                            new Actor() { Name = "Jack Nickolson", Birthdate = new DateTime(1937, 4, 22)},
+                            new Actor() { Name = "Leonardo DiCaprio", Birthdate = new DateTime(1974, 11, 11)},
+                            new Actor() { Name = "Matt Damon", Birthdate = new DateTime(1970, 8, 10)}
+                        }},
+                        new Film() { Author = "Gus Van Sant", Name = "Good Will Hunting", Actors = new List<Actor>() {
+                            new Actor() { Name = "Matt Damon", Birthdate = new DateTime(1970, 8, 10)},
+                            new Actor() { Name = "Robin Williams", Birthdate = new DateTime(1951, 8, 11)},
+                        }},
+                        new Book() { Author = "Stephen King", Name="Finders Keepers", Pages = 200},
+                        "Leonardo DiCaprio"
+                    };
+
+            Console.WriteLine("1) " + string.Join(",",data.Except(data.OfType<ArtObject>()).Select(_ => _).ToArray()));
+            Console.WriteLine("2) " + string.Join(",", data.OfType<Film>().SelectMany(film => film.Actors).Select(actor => actor.Name).Distinct().ToArray()));
+            Console.WriteLine("3) " + data.OfType<Film>().SelectMany(film => film.Actors).Where(actor => actor.Birthdate.Month == 8).Select(actor=>actor).Count());
+            Console.WriteLine("4) " + string.Join(",", data.OfType<Film>().SelectMany(film => film.Actors).OrderBy(actor => actor.Birthdate.Year).Take(2).Select(actor => actor.Name).ToArray()));
+            Console.WriteLine("5) "+ string.Join(",", data.OfType<Book>().GroupBy(_ => _.Author).Select(_ => new {Author = _.Key, Books = _.Count()})));
+            Console.WriteLine("6) "+ string.Join(",", data.OfType<ArtObject>().GroupBy(_ => _.Author).Select(_ => new {Author = _.Key, ArtObject = _.Count()})));
+            Console.WriteLine("7) " + data.OfType<Film>().SelectMany(_ => _.Actors).SelectMany(_ => _.Name).Distinct().Count());
+            Console.WriteLine("8) " + string.Join(",",data.OfType<Book>().OrderBy(_=>_.Author).ThenBy(_ => _.Pages).Select(_ => _.Name).ToList()));
```

Comment:
Why do you need 'ToList' here? You can just join the Enumerable - it's an excessive cast.